firmware/library/L1_Drivers/st7066u_test.cpp

```diff
@@ -0,0 +1,266 @@
+
+        // The driver should send the kTurnDisplayOff command when turning the
+        // display ON.
+        lcd.SetDisplayOn();
+        CHECK(!!(local_gpio[0].PIN & (1 << kRSPin)) == kCommandWrite);
+        CHECK(!!(local_gpio[0].PIN & (1 << kRWPin)) == kWriteOperation);
+        CHECK((local_gpio[1].PIN & 0xFF) == kDisplayOn);
+    }
+
+    SECTION("Show / Hide Cursor")
+    {
+        const uint8_t kShowCursor =
+            static_cast<uint8_t>(St7066u::Command::kTurnCursorOn);
+        const uint8_t kHideCursor =
+            static_cast<uint8_t>(St7066u::Command::kTurnDisplayOn);
+
+        St7066u lcd = St7066u(St7066u::BusMode::kEightBit,
+                              St7066u::DisplayMode::kSingleLine,
+                              St7066u::FontStyle::kFont5x8, &pins);
+        lcd.Initialize();
+
+        // The driver should send the kTurnCursorOn command to show the cursor.
+        lcd.SetCursorHidden(false);
+        CHECK(!!(local_gpio[0].PIN & (1 << kRSPin)) == kCommandWrite);
+        CHECK(!!(local_gpio[0].PIN & (1 << kRWPin)) == kWriteOperation);
+        CHECK((local_gpio[1].PIN & 0xFF) == kShowCursor);
+
+        // The driver should send the kTurnDisplayOn command to hide the cursor.
+        lcd.SetCursorHidden();
+        CHECK(!!(local_gpio[0].PIN & (1 << kRSPin)) == kCommandWrite);
+        CHECK(!!(local_gpio[0].PIN & (1 << kRWPin)) == kWriteOperation);
+        CHECK((local_gpio[1].PIN & 0xFF) == kHideCursor);
+    }
+
+    SECTION("Setting Cursor Direction")
+    {
+        constexpr uint8_t kCursorDirectionBackward =
+            static_cast<uint8_t>(St7066u::Command::kCursorDirectionBackward);
+        constexpr uint8_t kCursorDirectionForward =
+            static_cast<uint8_t>(St7066u::Command::kCursorDirectionForward);
+
+        St7066u lcd = St7066u(St7066u::BusMode::kEightBit,
+                              St7066u::DisplayMode::kMultiLine,
+                              St7066u::FontStyle::kFont5x8, &pins);
+        lcd.Initialize();
+
+        // The driver should write the command 0x04 to set the cursor
+        // direction to move in a forward direcion
+        lcd.SetCursorDirection(St7066u::CursorDirection::kBackward);
+        CHECK(!!(local_gpio[0].PIN & (1 << kRSPin)) == kCommandWrite);
+        CHECK(!!(local_gpio[0].PIN & (1 << kRWPin)) == kWriteOperation);
+        CHECK((local_gpio[1].PIN & 0xFF) == kCursorDirectionBackward);
+
+        // The driver should write the command 0x06 to set the cursor
+        // direction to move in a backwards direcion
+        lcd.SetCursorDirection(St7066u::CursorDirection::kForward);
+        CHECK(!!(local_gpio[0].PIN & (1 << kRSPin)) == kCommandWrite);
+        CHECK(!!(local_gpio[0].PIN & (1 << kRWPin)) == kWriteOperation);
+        CHECK((local_gpio[1].PIN & 0xFF) == kCursorDirectionForward);
+    }
+
+    SECTION("Setting Cursor Position")
+    {
+        constexpr uint8_t kLineNumber = 2;
+        constexpr uint8_t kPosition   = 18;  // 0x12
```

Comment:
I'm not sure, but if your idea was to put 0x12 into kPostition, I would recommend writing this as 0x12 rather than converting it to decimal.